doc/website/blog/2023-02-23-simple-cms-with-ent.mdx

```diff
@@ -0,0 +1,779 @@
+
+	// Read the response body and assert it contains "Hello, world!"
+	body, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+	require.Contains(t, string(body), "Hello, World!")
+}
+```
+
+Run the test to verify our server works correctly:
+
+```
+go test ./...
+```
+
+Observe our test passes:
+```
+ok      github.com/rotemtam/ent-blog-example    0.719s
+?       github.com/rotemtam/ent-blog-example/ent        [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/enttest        [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/hook   [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/migrate        [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/post   [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/predicate      [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/runtime        [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/schema [no test files]
+?       github.com/rotemtam/ent-blog-example/ent/user   [no test files]
+
+```
+
+#### Putting everything together
+
+Finally, let's update our `main` function to put everything together:
+
+```go
+func main() {  
+   // Read the connection string to the database from a CLI flag.  
+   var dsn string  
+   flag.StringVar(&dsn, "dsn", "", "database DSN")  
+   flag.Parse()  
+  
+   // Instantiate the Ent client.  
+   client, err := ent.Open("mysql", dsn)  
+   if err != nil {  
+      log.Fatalf("failed connecting to mysql: %v", err)  
+   }  
+   defer client.Close()  
+  
+   ctx := context.Background()  
+   // If we don't have any posts yet, seed the database.  
+   if !client.Post.Query().ExistX(ctx) {  
+      if err := seed(ctx, client); err != nil {  
+         log.Fatalf("failed seeding the database: %v", err)  
+      }  
+   }  
+   srv := newServer(client)  
+   r := newRouter(srv)  
+   log.Fatal(http.ListenAndServe(":8080", r))  
+}
+```
+
+We can now run our application and stand amazed at our achievement: a working blog front page!
+
+```
+ go run main.go -dsn "root:pass@tcp(localhost:3306)/test?parseTime=true"
+```
+
+![](https://entgo.io/images/assets/cms-blog/cms-01.png)
+
+### Step 4: Adding content
+
+:::info 
+You can follow the changes in this step in [this commit](https://github.com/rotemtam/ent-blog-example/commit/2e412ab2cda0fd251ccb512099b802174d917511).
+:::
+
+No content mangement system would be complete without the ability, well, to manage content. Let's demonstrate how we can add support for publishing new posts on our blog.
+
+Let's start by creating the backend handler:
+```go
+// add creates a new blog post.
+func (s *server) add(w http.ResponseWriter, r *http.Request) {
+	author, err := s.client.User.Query().Only(r.Context())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if err := s.client.Post.Create().
+		SetTitle(r.FormValue("title")).
+		SetBody(r.FormValue("body")).
```

Comment:
Calling `FormValue` without  `ParseForm` will result in ignoring errors from the implicit call to  `ParseForm`. https://pkg.go.dev/net/http#Request.FormValue

Reply:
That's a fair point, I hadn't considered it. But I think its a bit "nitpicking" in the scope of this post